### ISSUE TITLE
11:03:30 PM [vite-plugin-svelte] WARNING: The following packages use a svelte resolve configuration in package.json that has conflicting results and is going to cause problems future.  svelte-dnd-action@0.9.43  Please see https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#conflicts-in-svelte-resolve for details.

### ISSUE THREAD

**Issue by vlrevolution** [isaacHagoel/svelte-dnd-action#562]
Getting a warning:
11:03:30 PM [vite-plugin-svelte] WARNING: The following packages use a svelte resolve configuration in package.json that has conflicting results and is going to cause problems future.

svelte-dnd-action@0.9.43

Please see https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#conflicts-in-svelte-resolve for details.

is it fixable?

**Comment by trasherdk** [comment]
I get a warning, saying I have a resolve problem.

Can you please read the documentation, and fix my problem?

No, I can't share any configuration or package.json to help narrow the problem down.

**Comment by isaacHagoel** [comment]
@vlrevolution can you please provide a way to reproduce this error?

**Comment by Miniontoby** [comment]
I also get this warning.

It is because this package still has an `svelte` key in the root of the `package.json`:

> Using the `svelte` field in `package.json` to point at `.svelte` source files is deprecated and you must use a svelte [export condition](https://nodejs.org/api/packages.html#conditional-exports). vite-plugin-svelte 3 still resolves it as a fallback, but in a future major release this is going to be removed and without exports condition resolving the library is going to fail.
[Source](https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#missing-exports-condition)

So basically what vite wants you to do is to make use of the `exports` -> `.` -> `svelte` key. And I can see that you have already done that.

They also say you can keep the `svelte` field in addition to the exports condition for backwards compatibility, however they didn't write an check/if statement to see if you already have the exports condition, so it is basically no problem at the moment, but it is more like a bug at vite...

If you want to get rid of the warning you can remove the `svelte` field in the root of the `package.json` file inside the `node_modules/svelte-dnd-action/` folder.


----

Btw if you want to reproduce, just create a new project (the vite version I am using is `4.5.0`), add the package (import it also of course) and then run `vite build` and you will see the warning popping up:

```
PS C:\MyProjects\MyProject> npm run build

> MyProject@0.0.1 build
> vite build


vite v4.5.2 building SSR bundle for production...
✓ 123 modules transformed.
11:22:29 [vite-plugin-svelte] WARNING: The following packages use a svelte resolve configuration in package.json that has conflicting results and is going to cause problems future.

svelte-dnd-action@0.9.48

Please see https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#conflicts-in-svelte-resolve for details.

vite v4.5.2 building for production...
✓ 136 modules transformed.
11:22:34 [vite-plugin-svelte] WARNING: The following packages use a svelte resolve configuration in package.json that has conflicting results and is going to cause problems future.

svelte-dnd-action@0.9.48

Please see https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#conflicts-in-svelte-resolve for details.
.svelte-kit/output/client/_app/version.json                                  0.03 kB │ gzip:  0.05 kB
.svelte-kit/output/client/.vite/manifest.json                                6.94 kB │ gzip:  0.96 kB
(...)
```

**Comment by isaacHagoel** [comment]
Thanks

**Comment by Miniontoby** [comment]
I have reported it over at https://github.com/sveltejs/vite-plugin-svelte/issues/725

**Comment by Miniontoby** [comment]
Oh wait

You didn't make the two fields the EXACT same....
```
    "svelte": "src/index.js",
...
            "svelte": "./src/index.js"
```

**Comment by isaacHagoel** [comment]
there was a reason for that. I tried identical initially but it cause
another issue if I recall. I'll need to look at the commits/ PRs history
(will do later tonight or tomorrow)

**Comment by Miniontoby** [comment]
In the issue I created over at svelte-js, one dude actually also pointed this out:

> please also check your package type field, it looks like .js might currently be mistaken as commonjs.
> https://publint.dev/svelte-dnd-action


(yes, he thinks this repo is mine, but it isn't but yeah, just read it as it is directly for you...)

**Comment by Miniontoby** [comment]
Oke here is some information about the warning:

> This particular warning had been added in #510 and then removed again in #747 it is only present in vite-plugin-svelte@2. So the easiest way for you to avoid it is to upgrade to vite-plugin-svelte@3.

Seems like we should just upgrade to vite-plugin-svelte@3 on our own projects/packages!

**Comment by Miniontoby** [comment]
Oke, once you have updated, you, @isaacHagoel need to update the package.json of the package as well.

Please see https://github.com/sveltejs/vite-plugin-svelte/issues/725#issuecomment-2137191183

**Comment by Miniontoby** [comment]
Oke, forget that previous message with the link to the other issue!

**Comment by isaacHagoel** [comment]
@Miniontoby thanks for investigating this. I am not sure if I need to do something about this or not 😅

**Comment by Miniontoby** [comment]
> @Miniontoby thanks for investigating this. I am not sure if I need to do something about this or not 😅

Yeah, you don't have to do anything here. People who get this error just need to update their packages!

I will remove my other comments, to avoid more confusion for others, you may also remove the ones that you posted after my initial comment

**Comment by isaacHagoel** [comment]
Thanks. Done

**Comment by isaacHagoel** [comment]
Closing as a duplicate of the retained package-resolution issue #536. Packaging compatibility discussion should continue there.
